src/components/AnnotationViewer.vue

```diff
@@ -311,18 +288,20 @@ export default class AnnotationViewer extends Vue {
     this.clearOldAnnotations();
 
     // We want to ignore these already displayed annotations
-    const displayedIds = this.annotationLayer
+    const noRedrawGeoJSAnnotations = this.annotationLayer
       .annotations()
-      .map((a: any) => a.options("girderId"))
-      .filter(
-        (id: string) =>
-          id !== this.hoveredAnnotationId || !this.isAnnotationSelected(id)
-      );
+      .filter((geoJSAnnotation: any) => {
+        const id = geoJSAnnotation.options("girderId");
+        return !(
+          // Force redraw of annotations which are selected or hovered
+          (id === this.hoveredAnnotationId || this.isAnnotationSelected(id))
+        );
+      });
 
     // Then draw the new annotations
-    this.drawNewAnnotations(displayedIds);
+    this.drawNewAnnotations(noRedrawGeoJSAnnotations);
```

Comment:
your variable named "noRedraw.." is used in a method called "draw...". I think it makes the code a bit confusing...